frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlerts.tsx

```diff
@@ -60,18 +78,52 @@ export const MonitoringAlerts: React.FC<Props> = ({ match, rules, filters, listS
   const monitoringAlertColumn = React.useMemo(() => MonitoringAlertColumn(t), [t]);
   const columnIndex = _.findIndex(monitoringAlertColumn, { title: listSortBy });
   const sortOrder = listOrderBy || SortByDirection.asc;
-  const [response, loadError, loading] = usePrometheusRulesPoll({ namespace });
-  const thanosAlertsAndRules = React.useMemo(
-    () => (!loading && !loadError ? getAlertsAndRules(response?.data) : { rules: [], alerts: [] }),
-    [response, loadError, loading],
+  const [customExtensions] = useResolvedExtensions<AlertingRulesSourceExtension>(
+    isAlertingRulesSource,
+  );
+
+  const alertsSource = React.useMemo(
+    () =>
+      customExtensions
+        .filter((extension) => extension.properties.contextId === 'dev-observe-alerting')
+        .map((extension) => extension.properties),
+    [customExtensions],
   );
-  useAlertManagerSilencesDispatch({ namespace });
 
   React.useEffect(() => {
-    const sortThanosRules = _.sortBy(thanosAlertsAndRules.rules, alertingRuleStateOrder);
-    dispatch(alertingSetRules('devRules', sortThanosRules, 'dev'));
-    dispatch(alertingLoaded('devAlerts', thanosAlertsAndRules.alerts, 'dev'));
-  }, [dispatch, thanosAlertsAndRules]);
+    const url = getPrometheusURL({
+      endpoint: PrometheusEndpoint.RULES,
+      namespace,
+    });
+    dispatch(alertingLoading('alerts', 'dev'));
+
+    const poller = (): void => {
```

Comment:
The two `poller`s in this PR and the `poller` in `/components/monitoring/alerting.tsx` are very similar, so I think it would be good to combine them all.